ls1intum/Artemis
# `Assessment`: Extend example submission dashboard

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


Since the instructors are able to import an example submission using an existing student submission, it is hard to understand which submission is the imported one. This PR extends the example submission table with two information: result and submission size. Submission size differs depends on the exercise type. For the text exercises it shows number of the word in the text. For the modeling exercise, it shows the number of elements in the model (elements(for UML diagram class, attributes, methods) + associations). Also, the import modal shows the submission size.

### Description


- the table in `example-submission.component` extended (result+ submission size)
- the table in `example-submission-import.component` extended (submission size)
- related tests suites are covered and increased coverage
- to calculate the submission size, corresponding helper method is added in `example-submission.service`
- since the example submissions are not related with participant, so I had to add a condition in `result.component`
- I also noticed that, after saving the example submission result is not calculated and not stored in database. I added the missing part in `AssessmentResource`. 

note: Feel free to suggest correct translations in German!

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Modeling and 1 Text Exercise
- at least one student submission for  both of them (their assessment should be completed!!)

1. Log in to Artemis
2. Navigate to Course Management
3. Create or find existing modeling/text exercise
4. Go to Example Submissions
5. Create new one and assess it 
6. Check the example submissions table, you should see the result properly (no grading result - if there is no assessment)
7. Check the submission size in table also in the import modal (click import an Example Submission button)

**P.S. If you are testing using an existing example submission that has been evaluated before, probably you will se the result as 'No grading result'.  Before this implementation, the result was not saved properly. It was the missing part for example submissions. So, you need to open this existing example submission and click 'Show Assessment' then save the assessment, After that, you should see the the result.


### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






example-submissions.component.ts    |   84.78 
example-submission-import.component.ts  |   94.92 

### Screenshots


Before
![image](https://user-images.githubusercontent.com/23650630/141691373-d1194c1b-5135-4368-88d9-826e7c18dfb1.png)
After
![ssssss](https://user-images.githubusercontent.com/23650630/144271183-2ca2c0a4-44fb-40bb-bd1d-c898200439c9.jpg)

Before
![image](https://user-images.githubusercontent.com/23650630/141691392-911c0c8e-4fe8-4de5-b791-765ee20d195f.png)
After
![image](https://user-images.githubusercontent.com/23650630/144271419-eb28b168-2cdf-4360-b873-cef6a519c663.png)
